src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -631,11 +631,6 @@ public static synchronized void registerColumnEncryptionKeyStoreProviders(
                     0, false);
         }
 
-        if (null != globalCustomColumnEncryptionKeyStoreProviders) {
```

Comment:
I would propose doing it reverse way, as under:
```java
if (null == globalCustomColumnEncryptionKeyStoreProviders) {
    globalCustomColumnEncryptionKeyStoreProviders = new HashMap<>();
    // update the static Map
}
```

Reply:
I think for this release that will be fine, however I can see a problem down the road where we want to be running two different connections (with different Key Vaults) within the same JVM - the static nature would mean this would not be possible without some classloader hacking. However I also appreciate that's a much bigger change to the code base, and as we do not have that requirement right now I suggest we go with your change.

Reply:
@ccdigix 
Yes, for now you may merge the PR and we can limit this change to support single AKV Provider registration as currently driver works.

We are also following up internally for design changes needed to support multiple AKV providers, but its still in discussion, not yet confirmed.